test/Microsoft.Azure.WebJobs.Host.UnitTests/ExceptionFormatterTests.cs

```diff
@@ -0,0 +1,136 @@
+﻿using Microsoft.Azure.WebJobs.Host.Diagnostics;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace Microsoft.Azure.WebJobs.Host.UnitTests
+{
+    public class ExceptionFormatterTests
+    {
+        [Fact]
+        public void FormatException_RemovesAsyncFrames()
+        {
+            try
+            {
+                var test = new TestClass();
+                test.Run();
+            }
+            catch (Exception exc)
+            {
+                string formattedException = ExceptionFormatter.GetFormattedException(exc);
+
+                Assert.False(Regex.Match(formattedException, "d__.\\.MoveNext()").Success);
```

Comment:
it would help see what is actually going on if these tests tested a full string literal, so we can see the true formatting, and it would also help catch errors better. I realize though that line numbers might change over time, making this hard. Perhaps if the test class were in its own file, so line numbers don't jump around often? or perhaps if a regex is used to match any line numbers?